plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TrinoSqlExecutorWithRetries.java

```diff
@@ -0,0 +1,53 @@
+package io.trino.plugin.redshift;
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import dev.failsafe.Failsafe;
+import dev.failsafe.RetryPolicy;
+import io.trino.Session;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.SqlExecutor;
+
+import static com.google.common.base.Throwables.getCausalChain;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoSqlExecutorWithRetries
+        implements SqlExecutor
+{
+    private final QueryRunner queryRunner;
+    private final Session session;
+
+    public TrinoSqlExecutorWithRetries(QueryRunner queryRunner)
```

Comment:
Could instead wrap a SqlExecutor like in `ClickhouseSqlExecutor` to match a few more cases I put comments on that could be a source of flakes. Of course then could get a little odd that you might get a connection to trino, or a connection to redshift, and you'd be treating them with the same retry policy ...